src/test/java/com/mojang/brigadier/CommandDispatcherUsagesTest.java

```diff
@@ -100,14 +99,28 @@ public void setUp() throws Exception {
             literal("k")
                 .redirect(get("h"))
         );
+        //default node tests
+        subject.register(
+            literal("l")
+                //basic
+                .then(defaultLiteral("1").executes(command))
+                //two choices
+                .then(literal("2")
+                    .then(defaultLiteral("i").executes(command))
+                    .then(literal("ii").executes(command)))
+                //chained
+                .then(literal("3").then(defaultLiteral("i").then(defaultLiteral("ii").executes(command))))
+                //unexecutable default node
+                .then(literal("4").then(defaultLiteral("i").then(literal("ii").executes(command))))
+        );
     }
 
     private CommandNode<Object> get(final String command) {
-        return Iterables.getLast(subject.parse(command, source).getContext().getNodes()).getNode();
+        return Iterables.getLast(subject.parse(command, source).getContext().getNodes().stream().filter(n -> !n.getRange().isEmpty()).collect(Collectors.toList())).getNode();
     }
 
     private CommandNode<Object> get(final StringReader command) {
-        return Iterables.getLast(subject.parse(command, source).getContext().getNodes()).getNode();
+        return Iterables.getLast(subject.parse(command, source).getContext().getNodes().stream().filter(n -> !n.getRange().isEmpty()).collect(Collectors.toList())).getNode();
```

Comment:
Hmm, yeah.
It might be better if the `CommandContext` provided a method to get the actual last parsed node. It seems a bit verbose to me to try getting the last node that way, plus you could easily store a reference to that node in the command context, so it would be more efficient as well.
Of course, the code of projects using this library still would require changing, but that is only if default nodes are actually used and this new feature is "approved" by doing so. I'm not too worried about backwards compatibility for that reason.